estate/models/estate_property.py

```diff
@@ -0,0 +1,86 @@
+from odoo import api, fields, models
+from dateutil.relativedelta import relativedelta
+
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+from .helper import format_selection
+
+from odoo.fields import Many2many, One2many
+
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Estate Property modelisation'
+    _order = 'id desc'
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(default=lambda x: fields.Date.today() + relativedelta(months=3), copy=False)
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(string='Orientation',
+                                          selection=format_selection(['north', 'south', 'east', 'west']))
+    active = fields.Boolean(default=True)
+    state = fields.Selection(string='State',
+                             selection=format_selection(
+                                 ['new', 'offer received', 'offer accepted', 'sold', 'canceled']),
+                             default='new')
+
+    property_type_id = fields.Many2one('estate.property.type')
+
+    buyer = fields.Many2one('res.partner', copy=False)
+    salesperson = fields.Many2one('res.users', default=lambda self: self.env.user)
+
+    tag_ids = Many2many('estate.property.tag')
+
+    offer_ids = One2many('estate.property.offer', 'property_id')
+
+    total_area = fields.Integer(compute='_compute_total_area')
+
+    best_price = fields.Float(compute='_compute_best_price')
+
+    _sql_constraints = [
+        ('expected_price_strictly_positive', 'CHECK(expected_price > 0)', 'Expected price must be stricly positive'),
+        ('selling_price_positive', 'CHECK(selling_price >= 0)', 'Selling price must be positive'),
+    ]
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
```

Comment:
Use a more meaningful variable name. Trust me it'll be helpful later when you right big ass compute methods
```suggestion
        for estate in self:
            estate.total_area = estate.living_area + estate.garden_area
```